pulsar/producer_partition.go

```
@@ -184,6 +185,7 @@ func newPartitionProducer(client *client, topic string, options *ProducerOptions
 		options:          options,
 		producerID:       client.rpcClient.NewProducerID(),
 		dataChan:         make(chan *sendRequest, maxPendingMessages),
+		writeChan:        make(chan *pendingItem, maxPendingMessages),
```

This doesn't address the root cause. If we send too many messages, the `writeChan` can still get stuck, and the issue persists. I can reproduce the problem by producing at least `maxPendingMessages` after the cluster is shut down.

---

@RobertIndie Once the internal queue reaches its configured capacity (`maxPendingMessages`), the producer should either block or reject incoming messages based on the `DisableBlockIfQueueFull` setting:

- If `DisableBlockIfQueueFull` is **`false`**, the producer should **block** until space becomes available.
- If `DisableBlockIfQueueFull` is **`true`**, the producer should **fail fast** by rejecting new messages.

Proper backpressure handling is essential to avoid unbounded memory usage and ensure the system remains responsive under load or during broker unavailability.

By default, `DisableBlockIfQueueFull` is set to `false`, which means the producer will block when the number of pending messages exceeds `maxPendingMessages`. If it is explicitly set to true, the producer will fail fast with a producer send queue is full error.

I think this topic is different from the current PR and might be better addressed separately. Let’s keep this PR focused on the specific issue it aims to resolve.